pkg/storj/pieceid.go

```diff
@@ -66,11 +67,12 @@ func (id PieceID) String() string { return pieceIDEncoding.EncodeToString(id.Byt
 // Bytes returns bytes of the piece ID
 func (id PieceID) Bytes() []byte { return id[:] }
 
-// Derive a new PieceID from the current piece ID and the given storage node ID
-func (id PieceID) Derive(storagenodeID NodeID) PieceID {
+// Derive a new PieceID from the current piece ID, the given storage node ID and piece number
+func (id PieceID) Derive(storagenodeID NodeID, pieceNum int32) PieceID {
 	// TODO: should the secret / content be swapped?
 	mac := hmac.New(sha512.New, id.Bytes())
-	_, _ = mac.Write(storagenodeID.Bytes()) // on hash.Hash write never returns an error
+	_, _ = mac.Write(storagenodeID.Bytes())               // on hash.Hash write never returns an error
+	_, _ = mac.Write([]byte(strconv.Itoa(int(pieceNum)))) // on hash.Hash write never returns an error
```

Comment:
Instead of using strconv you can use little or big endian to write the data.
```
mac.Write([]byte{
    byte(uint32(piecenum) >> 0),
    byte(uint32(piecenum) >> 8),
    byte(uint32(piecenum) >> 16),
    byte(uint32(piecenum) >> 24),
})
```

Reply:
done